### ISSUE TITLE
A question about train.df and test.df

### ISSUE THREAD

**Issue by Jami1141** [Minyus/causallift#20]
Hello. 

I am using your code for my data set. But I have very urgent question, which I am confused now. 

As I understood, you mentioned that you train two separated models one for untreated people and one for treated people. After you mentioned that to calculate CATE, we combine all samples including treated and untreated people and then set treatment feature to 0 for all sample and give all sample to untreated model and obtain predicted probabilities, then another time we set for all sample treatment feature to 1 and give all sample to treated model and at the end we calculate the CATE subtracting predicted probabilities.
But in this case it means there are some cases which are seen for the treated model or untreated model. This is biased no?
 
I see that at the beginning of the code you split data in train and test set by 20%. But I don't understand where train and test set in two separated models come from. Is it related to train and test set we split at the beginning of the code? 

In any case, I am confused that If we train a model for untreated sample and train a model for treated sample, then we set treatment for all samples including treated and untreated to 0 and another time to 1 and give them two models separately, then some data are seen for one of the model. I am not sure my interpretation of your code is correct but If yes then we will have bias for calculating uplift.

I will be grateful If you reply to my question. Since I am currently using your code.

Many thanks

**Comment by Minyus** [comment]
I'm not sure whether I understand your question, but I would answer yes if you ask whether the methodology is biased or not.
Methodologies based on machine learning, including this package, uses bias. 
If the model is based on bias which is true in the past but not true in the future, then inference doesn't work, so some techniques are used to mitigate overfitting to the past data.
The methodology used in this package works better or worse than other methodologies depending on the data/situations, but all the methodologies based on machine learning are biased in my understanding.
